src/background_thread.c

```diff
@@ -364,7 +364,11 @@ background_thread_create(tsd_t *tsd, unsigned arena_ind) {
 	pre_reentrancy(tsd);
 	int err;
 	load_pthread_create_fptr();
-	if ((err = pthread_create(&info->thread, NULL,
+	/*
+	 * To avoid complications (besides reentrancy), create internal
```

Comment:
Oh if the app defines it's own pthread_create, and we call it here, then we have a path this way: app => malloc => app (pthread_create) => malloc (called by pthread lib). I had this with HHVM and felt we should just cut the step in the middle. Does this make sense to you?